packages/api/src/utils/stripe.js

```diff
@@ -0,0 +1,378 @@
+import Stripe from 'stripe'
+import { randomString } from './billing.js'
+
+/**
+ * @typedef {import('stripe').Stripe} StripeInterface
+ */
+
+/**
+ * @typedef {import("./billing-types").BillingService} BillingService
+ */
+
+/**
+ * @typedef {object} StripeComForBillingService
+ * @property {Pick<StripeInterface['paymentMethods'], 'attach'>} paymentMethods
+ * @property {Pick<StripeInterface['setupIntents'], 'create'>} setupIntents
+ * @property {Pick<StripeInterface['customers'], 'retrieve'|'update'>} customers
+ */
+
+/**
+ * A BillingService that uses stripe.com
+ */
+export class StripeBillingService {
+  /**
+   * @param {StripeComForBillingService} stripe
+   */
+  static create (stripe) {
+    return new StripeBillingService(stripe)
+  }
+
+  /**
+   * @protected
+   * @param {StripeComForBillingService} stripe
+   */
+  constructor (stripe) {
+    /** @type {StripeComForBillingService}  */
+    this.stripe = stripe
+    /**
+     * @type {BillingService}
+     */
+    const instance = this // eslint-disable-line
+  }
+
+  async getPaymentMethod (customerId) {
+    const response = await this.stripe.customers.retrieve(customerId, {
+      expand: ['invoice_settings.default_payment_method']
+    })
+    if (response.object !== 'customer') {
+      throw new Error('unable to get payment method')
+    }
+    if (response.deleted) {
+      throw new Error('unexpected response.deleted')
```

Comment:
Are these internal Errors or user input error?

Reply:
@vasco-santos this is not related to user error. These would be legitimately unexpected errors that might indicate something else was wrong in the stripe.com-hosted customer db. It shouldn't need to keep anyone up at night, but it would be nice to have a modest alarm if these errors ever do come up, and throwing explicitly will give the debugger (probably me) a stack trace when that happens. Technically this case can happen (the type system told me), but it shouldn't, and if it does it is probably something that should be escalated to someone like us

Reply:
I removed these two untyped `Errror`s. The first wasn't really needed. The second I added a specific `CustomerNotFound` type for, and added it to the function signature so tsc helps callers be ready for it or other errors. cd82b249164dbd8d373d653c9671f564bcba8e63

Reply:
that sounds good. Thanks for details :)